Added .gitignore, change README and write tests for modules in test.py

CarpetJesus opened the pull request.

Commit:
Added .gitignore, change README and write tests for modules in test.py

Ayantea1 commented on tests.py:
```diff
@@ -1,24 +1,128 @@
-# класс TestBooksCollector объединяет набор тестов, которыми мы покрываем наше приложение BooksCollector
-# обязательно указывать префикс Test
-class TestBooksCollector:
+@pytest.fixture
```
можно улучшить: фикстуру лучше вынести в отдельный модуль

Ayantea1 commented on tests.py:
```diff
@@ -1,24 +1,128 @@
+def collector():
+    return BooksCollector()
+
+@pytest.fixture
```
Нужно исправить: начальное состояние списков задано в main

Ayantea1 commented on tests.py:
```diff
@@ -1,24 +1,128 @@
+    def test_set_book_genre_valid(self, collector):
+        collector.add_new_book("Темная Башня")
+        collector.set_book_genre("Темная Башня", "Фантастика")
+        assert collector.get_book_genre("Темная Башня") == "Фантастика"
```
Нужно исправить: для get_book_genre должен быть отдельный тест. Тесты не должны дублировать друг друга. В идеале - чтобы в каждом тесте использовался только один проверяемый метод, все остальное можно получить прямым доступом к словарям и спискам

Ayantea1 commented on tests.py:
```diff
@@ -1,24 +1,128 @@
+        collector.add_new_book("Властелин Колец")
+        collector.set_book_genre("Темная Башня", "Фантастика")
+        collector.set_book_genre("Властелин Колец", "Фантастика")
+        assert collector.get_books_with_specific_genre("Фантастика") == ["Темная Башня", "Властелин Колец"]
```
можно улучшить: метод должен выбирать книги с нужным жанром, а в этом тесте возвращаются все

Commit:
- Убрал фикстуру в отдельный модуль
- Я совсем не понял комментария про main. Решил исправить так: Убрал две фикстуры со списками и переделал тесты с инит, добавив в них переменные со списками, внутри теста. Прогнал по циклу и сделал assert
- Убрал из тестов с set_book_genre проверку через get_book_genre. Переделал в проверку через ключ словаря
- Сделал отдельные тесты для get_book_genre.
- Переделал проверку для test_get_books_with_specific_genre. Чтобы проверка была добавления книг именно с заданым жанром
